Mathlib/Topology/Algebra/Order/LeftRight.lean

```diff
@@ -27,6 +27,46 @@ left continuous, right continuous
 
 open Set Filter Topology
 
+section Preorder
+
+variable {α : Type*} [TopologicalSpace α] [Preorder α]
+
+lemma frequently_lt_nhds (a : α) [NeBot (𝓝[<] a)] : ∃ᶠ x in 𝓝 a, x < a :=
+  frequently_iff_neBot.2 ‹_›
+
+lemma frequently_gt_nhds (a : α) [NeBot (𝓝[>] a)] : ∃ᶠ x in 𝓝 a, a < x :=
+  frequently_iff_neBot.2 ‹_›
+
+theorem Filter.Eventually.exists_lt {a : α} [NeBot (𝓝[<] a)] {p : α → Prop}
+    (h : ∀ᶠ x in 𝓝 a, p x) : ∃ b < a, p b :=
+  ((frequently_lt_nhds a).and_eventually h).exists
+#align filter.eventually.exists_lt Filter.Eventually.exists_lt
+
+theorem Filter.Eventually.exists_gt {a : α} [NeBot (𝓝[>] a)] {p : α → Prop}
+    (h : ∀ᶠ x in 𝓝 a, p x) : ∃ b > a, p b :=
+  ((frequently_gt_nhds a).and_eventually h).exists
+#align filter.eventually.exists_gt Filter.Eventually.exists_gt
+
+theorem nhdsWithin_Ici_neBot {a b : α} (H₂ : a ≤ b) : NeBot (𝓝[Ici a] b) :=
+  nhdsWithin_neBot_of_mem H₂
+#align nhds_within_Ici_ne_bot nhdsWithin_Ici_neBot
+
+@[instance]
```

Comment:
Do you understand why this a `theorem` with `@[instance]` instead of just `instance`?

Reply:
It looks like I did it at the same time I changed it to use `ne_bot` in Lean 3 times. I don't remember why.